api/src/test/java/org/apache/iceberg/transforms/TestResiduals.java

```diff
@@ -214,6 +218,36 @@ public void testInTimestamp() {
     Assert.assertEquals("Residual should be alwaysFalse", alwaysFalse(), residual);
   }
 
+  @Test
+  public void testInNaN() {
```

Comment:
Do we need NaN cases for other evaluators?

Reply:
You mean testing `InNaN` case for other evaluators? Yeah I'll do that

Reply:
Since we decided that we don't accept NaN in `in`, I guess this conversation is outdated. I've removed `inNaN` test from here.